### ISSUE TITLE
\0 Byte in JUnit tests causes PIT to create invalid XML

### ISSUE THREAD

**Issue by uhafner** [hcoles/pitest#468]
In one of my [unit tests](https://github.com/jenkinsci/analysis-model/commit/620477fb2f05ec14e9e9c9a2432857e6b5d9f187) I'm verifying that the 0 byte is correctly handled. While the test passes, PIT (pitest-maven:1.4.0:mutationCoverage) creates an invalid XML file.

``` 
<mutation detected='true' status='KILLED' numberOfTestsRun='4'>
<sourceFile>FingerprintGenerator.java</sourceFile>
<mutatedClass>edu.hm.hafner.analysis.FingerprintGenerator</mutatedClass>
<mutatedMethod>run</mutatedMethod>
<methodDescription>(Ledu/hm/hafner/analysis/FullTextFingerprint;Ledu/hm/hafner/analysis/Report;Ljava/nio/charset/Charset;)V</methodDescription>
<lineNumber>35</lineNumber><mutator>org.pitest.mutationtest.engine.gregor.mutators.VoidMethodCallMutator</mutator>
<index>48</index>
<block>4</block>
<killingTest>edu.hm.hafner.analysis.FingerprintGeneratorTest.[3] Illegal filename = &#0; Null-Byte</killingTest>
<description>removed call to edu/hm/hafner/analysis/Issue::setFingerprint</description>
</mutation>
```

**Comment by maxgabut** [comment]
@uhafner #477 is a way of solving this problem. WDYT ?

**Comment by uhafner** [comment]
Looks good to me, thanks for fixing!

**Comment by maxgabut** [comment]
Great ! I guess you can close the issue now.

**Comment by hcoles** [comment]
@maxgabut Yes, thanks for the PR. I hope to push a release out soon, just need to see if I can find time to look at the other outstanding PR first.

**Comment by hcoles** [comment]
Fixed in 1.4.1
